auto_invoice_advance/models/sale_order.py

```diff
@@ -9,32 +9,99 @@
 
+        # Get the configurations
         percentage_amount = self.env['ir.config_parameter'].sudo().get_param('sale.deposit_default_percentage')
+        confirm_down_payment = self.env['ir.config_parameter'].sudo().get_param('sale.auto_confirm_down_payment')
+        do_send_email = self.env['ir.config_parameter'].sudo().get_param('sale.auto_email_down_payment')
+
         down_payments_to_create = []
         for order_id in order_ids:
             down_payments_to_create.append(self._prepare_down_payment_values(order_id, percentage_amount))
-        self._create_advance_invoices(down_payments_to_create)
+        self._create_advance_invoices(down_payments_to_create,
+                                      confirm_down_payment,
+                                      do_send_email)
 
-    def _create_advance_invoices(self, down_payments_to_create):
+    def _create_advance_invoices(self, down_payments_to_create, confirm_down_payment, do_send_email):
         payment_inv_ids = self.env['sale.advance.payment.inv'].create(down_payments_to_create)
+        invoice_confirmable_ids = self.env['account.move']
         for payment_inv_id in payment_inv_ids:
-            payment_inv_id.create_invoices()  # singleton check in code, so unable to bulk process
-            self.env.cr.commit()
+            response = payment_inv_id.create_down_payment_invoices()
+            invoice_id = response.get('invoice_id')
+
+            if response.get('can_be_confirmed'):
+                invoice_confirmable_ids += invoice_id
+
+            # Link the created down payment invoice to the sale order(s)
+            # Note that there can be multiple sale orders on a single invoice, link all of them to the invoice
+            for sale_order_id in payment_inv_id.sale_order_ids:
+                sale_order_id.down_payment_id = invoice_id
+
+        if confirm_down_payment:
+            invoice_confirmable_ids.action_post()
+            if do_send_email:
+                invoice_confirmable_ids.print_and_send_automatic_emails()
+
+        # Done with our batch, keep it
+        self.env.cr.commit()
         _logger.info(f'Created {len(payment_inv_ids)} advance invoices')
 
     def _prepare_order_deposit_domain(self):
         return [('state', '=', 'sale'),         # For all the confirmed sales
                 ('invoice_ids', '=', None)]     # Without any invoices
 
     def _prepare_down_payment_values(self, order_id, percentage_amount):
+        # We will only allow the automatic confirming of invoices with one tax
+        deposit_taxes_id = self._try_get_tax_id(order_id)
+
         return {
             'sale_order_ids': order_id,
             'advance_payment_method': 'percentage',
-            'amount': percentage_amount
+            'amount': percentage_amount,
+            'deposit_taxes_id': deposit_taxes_id,
         }
+
+    def _try_get_tax_id(self, order_id):
```

Comment:
This code/whole function is horrific, sorry.
1. Your if `len()` check makes no sense. Just do a for loop over `order.order_line`. If there's no lines, there's nothing to loop, simple.
2. Create a variable `tax_ids` like this: `tax_ids = []` and in the loop check if the the tax_id(s) set on the line are not yet in the list. If not `append()` to the list. You will know all taxes on the lines and if there's none, one or multiple taxes.
3. Remove that whole try/catch stuff, not clean. 
4. Remove that `ensure_one`, you can have multiple taxes on one line in some cases so it makes no sense.